libnavigation-util/src/test/java/com/mapbox/navigation/utils/timer/MapboxTimerTest.kt

```diff
@@ -1,94 +1,121 @@
-        val testLambda = { lambdaCalled = true }
+    fun `should update interval dynamically`() = coroutineRule.runBlockingTest {
+        mapboxTimer.restartAfter = TimeUnit.MINUTES.toMillis(1)
+
+        val job = mapboxTimer.startTimer(mockLambda)
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(1))
 
-        val timer = MapboxTimer(100L, testLambda)
-        timer.start()
-        timer.stop()
-        advanceTimeBy(150)
+        mapboxTimer.restartAfter = TimeUnit.MINUTES.toMillis(10)
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(10))
 
-        assertFalse(lambdaCalled)
+        job.cancelAndJoin()
+        coVerify(exactly = 2) { mockLambda.invoke() }
     }
 
     @Test
-    fun executeLambda_notCalled_when_startNotCalled() = runBlockingTest {
-        var lambdaCalled = false
-        val testLambda = { lambdaCalled = true }
+    fun `should stop when canceled`() = coroutineRule.runBlockingTest {
+        mapboxTimer.restartAfter = TimeUnit.MINUTES.toMillis(5)
 
-        MapboxTimer(0L, testLambda)
-        advanceTimeBy(10)
+        val job = mapboxTimer.startTimer(mockLambda)
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(6))
 
-        assertFalse(lambdaCalled)
+        mapboxTimer.stopJobs()
+        coroutineRule.testDispatcher.advanceTimeBy(TimeUnit.MINUTES.toMillis(30))
+
+        job.cancelAndJoin()
+        coVerify(exactly = 1) { mockLambda.invoke() }
     }
 
     @Test
-    fun timerNotStartedUntilStartCalled() = runBlockingTest {
-        var counter = 0
-        val testLambda = { counter += 1 }
+    fun `should subscribe to jobs separately`() = coroutineRule.runBlockingTest {
+        val mockLambda1: () -> Unit = mockk(relaxed = true)
+        val mockLambda2: () -> Unit = mockk(relaxed = true)
+        mapboxTimer.restartAfter = 5000
+
+        val job1 = mapboxTimer.startTimer(mockLambda1)
+        coroutineRule.testDispatcher.advanceTimeBy(5001)
 
-        MapboxTimer(100L, testLambda)
-        advanceTimeBy(200L)
+        val job2 = mapboxTimer.startTimer(mockLambda2)
+        coroutineRule.testDispatcher.advanceTimeBy(5001)
 
-        assertEquals(0, counter)
+        job1.cancelAndJoin()
+        job2.cancelAndJoin()
+        coVerify(exactly = 2) { mockLambda1.invoke() }
+        coVerify(exactly = 1) { mockLambda2.invoke() }
     }
 
     @Test
-    fun multipleStartCalls_doNotExecuteMultipleJobs() = runBlocking {
-        var counter = 0
-        val testLambda = { counter += 1 }
-
-        val timer = MapboxTimer(100L, testLambda)
-        timer.start()
-        timer.start()
-        timer.start()
-        timer.start()
-        delay(150L)
-
-        assertEquals(1, counter)
+    fun `should stop and restart normally`() = coroutineRule.runBlockingTest {
+        mapboxTimer.restartAfter = TimeUnit.MINUTES.toMillis(5)
+
+        // Should emit 1
+        val job1 = mapboxTimer.startTimer(mockLambda)
```

Comment:
NIT not a big fan of naming like this `job1`, `job2`, etc. 😬 but not a big deal, could we come with a better name?

Reply:
hmm yeah i'm typically not either but this 1 means first. not sure if i rename it to first, if that seems passive to your comment. would you have a suggestion?

- [ ] jobFirst, jobSecond
- [ ] firstTimerJob, secondTimerJob

Reply:
> firstJob, secondJob

Sounds good to me 🚀